quickstart/daml/licensing-tests/daml/Licensing/Scripts/TestLicense.daml

```diff
@@ -0,0 +1,217 @@
+  -- User submits the app install request using the curl command displayed on
+  -- provider's Web UI against users' validator node using the JWT token from
+  -- the users' IAM
+  requestId <- submit user.primaryParty do
+    createCmd AppInstallRequest with
+      dso       = app.dso
+      provider  = app.provider.primaryParty
+      user      = user.primaryParty
+      meta      = emptyMetadata
+
+  -- A trigger running on the provider backend will pick up the request and
+  -- accept it provided the user party is known and there is no install yet
+  -- for the party. It also associated the provider user-id with the install
+  -- for disambiguation purposes.
+  void $ submit app.provider.primaryParty do
+    exerciseCmd requestId AppInstallRequest_Accept with
+      meta        = emptyMetadata
+      installMeta = Metadata with
+        values = Map.fromList [("providerUserId", "<user-id>")]
+
+-- | Simulate the setup of the validator party on a self-hosted validator node.
+setupValidatorParty : AmuletApp -> Text -> Script AmuletUser
+setupValidatorParty app name = do
+  party  <- allocatePartyWithHint name $ PartyIdHint name
+  userId <- validateUserId name
+  -- We need 'CanReadAs app.dso' only in tests, as they are not yet setup to simulate explicit disclosure.
+  -- In the actual implementation, the provider backend would provide the DSO published contracts via explicit disclosure.
+  createUser (User userId (Some party)) [CanActAs party, CanReadAs app.dso]
+  -- the validator party is its own validator and can thus claim all its validator rewards
+  recordValidatorOf app party party
+  pure $ AmuletUser userId party
+
+
+setupLicense : LicensingApp -> AmuletUser -> Script (ContractId License)
+setupLicense app user = do
+    installId <- getInstall
+    -- need: readAs dso for explicit disclosure of AmuletRules and OpenMiningRound
+    result <- submitMulti [app.provider.primaryParty] [app.dso] do
+      exerciseCmd installId AppInstall_CreateLicense with
+        params = licenseParams
+    pure result.licenseId
+  where
+    getInstall : Script (ContractId AppInstall)
+    getInstall = do
+      [(installId, _)] <- query @AppInstall user.primaryParty
+      pure installId
+    licenseParams = LicenseParams with
+      meta = Metadata with
+        values = Map.fromList [("licenseId", "<dummy-uuid>")]
+
+renewLicense : LicensingApp -> ContractId License -> Decimal -> RelTime -> Script (ContractId LicenseRenewalRequest, ContractId AppPaymentRequest)
+renewLicense app licenseId licenseFeeCc licenseExtensionDuration = submit app.provider.primaryParty do
+  exerciseCmd licenseId License_Renew with
+    licenseFeeCc
+    licenseExtensionDuration
+    paymentAcceptanceDuration = days 1
+    description = "Renewal of the license"
+
+
+completeTransfer : AmuletApp -> AmuletUser -> ContractId License -> ContractId AcceptedAppPayment -> Script ()
+completeTransfer app provider licenseCid acceptedPaymentCid = do
```

Comment:
```suggestion
completeTransfer app provider licenseCid reference = do
```
change to pass in the reference here, and use that to bind what's queried and used.